From 1037771dd5df79cd561099b395102bdd2497d441 Mon Sep 17 00:00:00 2001
From: Yang Shi <yang.shi@windriver.com>
Date: Fri, 16 Dec 2011 11:24:42 -0800
Subject: [PATCH 167/238] octeon/net: Use preempt safe smp_processor_id for threading interrupt

smp_processor_id is not preempt safe for threading interrupt, so change
to raw_smp_processor_id.

Signed-off-by: Yang Shi <yang.shi@windriver.com>
---
 drivers/net/octeon/ethernet-rx.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/net/octeon/ethernet-rx.c b/drivers/net/octeon/ethernet-rx.c
index 1f65383..301df69 100644
--- a/drivers/net/octeon/ethernet-rx.c
+++ b/drivers/net/octeon/ethernet-rx.c
@@ -111,7 +111,7 @@ static void cvm_oct_enable_one_cpu(void)
 			cvm_oct_napi[cpu].available--;
 			core_state.active_cores++;
 			spin_unlock_irqrestore(&core_state.lock, flags);
-			if (cpu == smp_processor_id()) {
+			if (cpu == raw_smp_processor_id()) {
 				cvm_oct_enable_napi();
 			} else {
 #ifdef CONFIG_SMP
-- 
1.7.0

